lib/samples/show_line_of_sight_between_points/show_line_of_sight_between_points.dart

```diff
@@ -0,0 +1,170 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import 'package:arcgis_maps/arcgis_maps.dart';
+import 'package:arcgis_maps_sdk_flutter_samples/common/common.dart';
+import 'package:flutter/material.dart';
+
+class ShowLineOfSightBetweenPoints extends StatefulWidget {
+  const ShowLineOfSightBetweenPoints({super.key});
+
+  @override
+  State<ShowLineOfSightBetweenPoints> createState() =>
+      _ShowLineOfSightBetweenPointsState();
+}
+
+class _ShowLineOfSightBetweenPointsState
+    extends State<ShowLineOfSightBetweenPoints>
+    with SampleStateSupport {
+  // Create a controller for the scene view.
+  final _sceneViewController = ArcGISSceneView.createController();
+
+  // The LocationLineOfSight object that will provide line-of-sight analysis for this sample.
+  // The object is initialized with the starting observer and target locations.
+  final _locationLineOfSight = LocationLineOfSight(
+    observerLocation: ArcGISPoint(
+      x: -73.095827750063904,
+      y: -49.319214695380957,
+      z: 2697.4689045762643,
+      spatialReference: SpatialReference.wgs84,
+    ),
+    targetLocation: ArcGISPoint(
+      x: -73.125568047959803,
+      y: -49.347049722534479,
+      z: 1944.1079967124388,
+      spatialReference: SpatialReference.wgs84,
+    ),
+  );
+
+  // A flag for when the scene view is ready.
+  var _ready = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: SafeArea(
+        top: false,
+        left: false,
+        right: false,
+        child: Stack(
+          children: [
+            Column(
+              children: [
+                Expanded(
+                  // Add a scene view to the widget tree and set a controller.
+                  child: ArcGISSceneView(
+                    controllerProvider: () => _sceneViewController,
+                    onSceneViewReady: onSceneViewReady,
+                    onTap: onTap,
+                    onLongPressEnd: onLongPressEnd,
+                  ),
+                ),
+                const Column(
+                  mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                  children: [
+                    Text('Tap to set new observation point.'),
+                    Text('Long press to set new target point.'),
+                    Divider(),
+                    Text('Green: Visible from the observation point.'),
+                    Text('Red: Not visible from the observation point.'),
+                    Text('Hidden: Segment is obscured by terrain.'),
+                  ],
+                ),
+              ],
+            ),
+            // Display a progress indicator and prevent interaction until state is ready.
+            LoadingIndicator(visible: !_ready),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Future<void> onSceneViewReady() async {
```

Comment:
This could just be `void onSceneViewReady() {` since there's no await'ing going on.